enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/interpreted/PrimitiveExecutionContext.scala

```diff
@@ -24,29 +24,33 @@ import org.neo4j.cypher.internal.frontend.v3_3.InternalException
 import org.neo4j.values.AnyValue
 
 object PrimitiveExecutionContext {
-  def empty = new PrimitiveExecutionContext(new PipelineInformation(Map.empty, 0,0))
+  def empty = new PrimitiveExecutionContext(new PipelineInformation(Map.empty, 0, 0))
 }
 
 case class PrimitiveExecutionContext(pipeline: PipelineInformation) extends ExecutionContext {
 
   private val longs = new Array[Long](pipeline.numberOfLongs)
   private val refs = new Array[AnyValue](pipeline.numberOfReferences)
 
-  def copyFrom(input: ExecutionContext, longsToCopy: Int): Unit = input match {
+  override def copyTo(target: ExecutionContext): Unit = target match {
     case other@PrimitiveExecutionContext(otherPipeline) =>
-      System.arraycopy(other.longs, 0, longs, 0, longsToCopy)
-      System.arraycopy(other.refs, 0, refs, 0, otherPipeline.numberOfReferences)
-
+      if (pipeline.numberOfLongs > otherPipeline.initialNumberOfLongs ||
```

Comment:
this should look at `otherPipeline.numberOfLongs`, and not `initialNumberOfLongs`, right?